lib/value.js

```diff
@@ -23,7 +23,7 @@ module.exports = function (valueType, value) {
     decode: function decode (target, offset, end) {
       offset = offset | 0
       if ((target.length - offset) < encodeLength) throw new RangeError('not enough data for decode')
-      if (valueBuffer.compare(target, offset, end) !== 0) throw new TypeError('Expected value ' + value)
+      if (valueBuffer.compare(target, offset, offset + encodeLength) !== 0) throw new TypeError('Expected value ' + value)
```

Comment:
This fix is sorely needed... but I wonder if I should be checking that `end - offset` is `>= encodeLength`